### PR TITLE
fix: support escaped string

### PR DESCRIPTION
**Pull request by yjhmelody**

### CODE REVIEW

**Review by koushiro — APPROVED** [review]

File: packages/as-serde-json/assembly/serializer.ts [hunk 1]
```diff
@@ -109,10 +109,37 @@ export class JSONSerializer extends Serializer<StringBuffer> {
...
+            }
+            else if(c =='\n') {
+                buf.write('\\n');
+            }
+            else if(c =='\r') {
+                buf.write('\\r');
+            }
+            else if(c =='\t') {
+                buf.write('\\t');
+            } else {
```
Comment (iamizumi) [inline packages/as-serde-json/assembly/serializer.ts:137]: it's better to fmt the code to use the same style for the if-else statements

File: packages/as-serde-json/assembly/serializer.ts [hunk 2]
```diff
@@ -109,10 +109,31 @@ export class JSONSerializer extends Serializer<StringBuffer> {
...
     serializeString(value: string): StringBuffer {
-        // TODO: escape
-        value = value.replace('"', '\\"');
+        let buf = new StringBuffer(value.length * 2);
+        for (let i = 0; i < value.length; i++) {
+            const c = value.slice(i, i + 1);
+            if (c == '\\') {
+                buf.write('\\\\');
+            } else if (c == '/') {
+                buf.write('\\/');
```
Comment (koushiro) [inline packages/as-serde-json/assembly/serializer.ts:118]:
```suggestion
                buf.write('\/');
```

Comment (iamizumi) [inline packages/as-serde-json/assembly/serializer.ts:118]:
```suggestion
                buf.write('\/');
```

File: packages/as-serde-json/assembly/serializer.ts [hunk 3]
```diff
@@ -109,10 +109,31 @@ export class JSONSerializer extends Serializer<StringBuffer> {
...
+            } else if (c == '"') {
+                buf.write('\\"');
```
Comment (iamizumi) [inline packages/as-serde-json/assembly/serializer.ts:120]:
```suggestion
                buf.write('\"');
```

File: packages/as-serde-json/assembly/serializer.ts [hunk 4]
```diff
@@ -109,10 +109,31 @@ export class JSONSerializer extends Serializer<StringBuffer> {
 
     @inline
     serializeString(value: string): StringBuffer {
-        // TODO: escape
-        value = value.replace('"', '\\"');
+        let buf = new StringBuffer(value.length * 2);
+        for (let i = 0; i < value.length; i++) {
+            const c = value.slice(i, i + 1);
+            if (c == '\\') {
+                buf.write('\\\\');
```
Comment (iamizumi) [inline packages/as-serde-json/assembly/serializer.ts:116]:
```suggestion
                buf.write('\\');
```

File: packages/as-serde-json/assembly/__tests__/testdata.ts [hunk 5]
```diff
@@ -76,8 +76,8 @@ export class SkipSuperNumbers extends Numbers {
 @serialize()
 export class Strings {
     s1: string = "";
-    s2: string = '"';
-    s3: string = "\r\n";
+    s2: string = '"/';
```
Comment (iamizumi) [inline packages/as-serde-json/assembly/__tests__/testdata.ts:79]:
```suggestion
    s2: string = '"/\\';
```

File: packages/as-serde-json/assembly/__tests__/serializer.spec.ts [hunk 6]
```diff
@@ -142,7 +142,7 @@ describe("JSONSerializer", () => {
 
     it("Strings", () => {
         let res = JSONSerializer.serialize(new Strings());
-        const expected = '{"s1":"","s2":"\\"","s3":"\r\n"}';
+        const expected = `{"s1":"","s2":"\\"\\/","s3":"\\b\\f\\n\\r\\t"}`;
```
Comment (iamizumi) [inline packages/as-serde-json/assembly/__tests__/serializer.spec.ts:145]:
```suggestion
        const expected = `{"s1":"","s2":"\"\/\\","s3":"\\b\\f\\n\\r\\t"}`;
```

Comment (iamizumi) [reply packages/as-serde-json/assembly/__tests__/serializer.spec.ts:145]:
```rust
fn main() {
    let s2 = r#""/\"#;
    let json = serde_json::to_string(&s2).unwrap();
    println!("{}", json);
}
```

result: `"\"/\\"`

File: packages/as-serde-json/assembly/serializer.ts (hunk 2 above)
Comment (iamizumi) [inline packages/as-serde-json/assembly/serializer.ts:118]:
```suggestion
```

File: packages/as-serde-json/assembly/serializer.ts (hunk 3 above)
Comment (iamizumi) [inline packages/as-serde-json/assembly/serializer.ts:120]:
```suggestion
                buf.write('\"');
```

**Review by lambda-world — APPROVED** [review]

**Pull request merged**